### PR TITLE
Change Java Class File version

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/41357

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/split/constants/JvmModuleConstantsGen.java [hunk 1]
```diff
@@ -99,7 +98,7 @@ public void generateConstantInit(Map<String, byte[]> jarEntries) {
     private void visitModuleField(ClassWriter cw, String varName) {
 
         FieldVisitor fv;
-        fv = cw.visitField(ACC_PUBLIC + ACC_FINAL + ACC_STATIC, varName, GET_MODULE, null, null);
+        fv = cw.visitField(ACC_PUBLIC + ACC_STATIC, varName, GET_MODULE, null, null);
```
[the remark this answers is not in the document]
Comment (warunalakshitha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/split/constants/JvmModuleConstantsGen.java:101]: Got following error after changing.
Update to static final field ballerina.lang$0046annotations.0.constants.$_string_constants_0.$bString0 attempted from a different method ($string_init) than the initializer method <clinit>

Seems java 17 code analyser checks any final variable changes at runtime.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmValueGen.java [hunk 2]
```diff
@@ -369,7 +368,7 @@ private byte[] createRecordValueClass(BRecordType recordType, String className,
                                                jvmPackageGen.symbolTable);
         JvmCastGen jvmCastGen = new JvmCastGen(jvmPackageGen.symbolTable, jvmTypeGen, types);
         LambdaGen lambdaGen = new LambdaGen(jvmPackageGen, jvmCastGen);
-        cw.visit(V1_8, ACC_PUBLIC + ACC_SUPER, className,
+        cw.visit(V17, ACC_PUBLIC + ACC_SUPER, className,
                 RECORD_VALUE_CLASS,
                 MAP_VALUE_IMPL, new String[]{MAP_VALUE});
```
Comment (Nadeeshan96) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmValueGen.java:373]:
```suggestion
        cw.visit(V17, ACC_PUBLIC + ACC_SUPER, className, RECORD_VALUE_CLASS, MAP_VALUE_IMPL, new String[]{MAP_VALUE});
```

**Review by gabilang — DISMISSED** [review]
LGTM

**Review by Nadeeshan96 — APPROVED** [review]
LGTM

**Pull request merged**